src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplication.groovy

```diff
@@ -18,6 +18,17 @@ package com.bmuschko.gradle.docker
 class DockerJavaApplication {
     String baseImage = 'java'
     String maintainer = System.getProperty('user.name')
+    @Deprecated
     Integer port = 8080
+    Integer[] ports = []
     String tag
+
+    Integer[] getPorts() {
```

Comment:
Lets change the implemenation to something like the below.

    Integer[] getPorts() {
        def openedPorts = ports.clone()
        if (port && !openedPorts.contains(port)) { openedPorts.add(port) }
        openedPorts
    }